AppCenterDataStorage/AppCenterDataStorage/Internal/LocalStore/MSDBDocumentStore.m

```diff
@@ -66,14 +89,17 @@ - (BOOL)deleteUserStorageWithAccountId:(NSString *)accountId {
 + (MSDBColumnsSchema *)columnsSchema {
 
   // TODO create composite key for partition and the document id
-  return @[
-    @{kMSIdColumnName : @[ kMSSQLiteTypeInteger, kMSSQLiteConstraintPrimaryKey, kMSSQLiteConstraintAutoincrement ]},
-    @{kMSPartitionColumnName : @[ kMSSQLiteTypeText, kMSSQLiteConstraintNotNull ]},
-    @{kMSDocumentIdColumnName : @[ kMSSQLiteTypeText, kMSSQLiteConstraintNotNull ]}, @{kMSDocumentColumnName : @[ kMSSQLiteTypeText ]},
-    @{kMSETagColumnName : @[ kMSSQLiteTypeText ]}, @{kMSExpirationTimeColumnName : @[ kMSSQLiteTypeInteger ]},
-    @{kMSDownloadTimeColumnName : @[ kMSSQLiteTypeInteger ]}, @{kMSOperationTimeColumnName : @[ kMSSQLiteTypeInteger ]},
-    @{kMSPendingDownloadColumnName : @[ kMSSQLiteTypeText ]}
-  ];
+  NSMutableArray *schema = [NSMutableArray new];
+  [schema addObject:@{kMSIdColumnName : @[ kMSSQLiteTypeInteger, kMSSQLiteConstraintPrimaryKey, kMSSQLiteConstraintAutoincrement ]}];
+  [schema addObject:@{kMSPartitionColumnName : @[ kMSSQLiteTypeText, kMSSQLiteConstraintNotNull ]}];
+  [schema addObject:@{kMSDocumentIdColumnName : @[ kMSSQLiteTypeText, kMSSQLiteConstraintNotNull ]}];
+  [schema addObject:@{kMSDocumentColumnName : @[ kMSSQLiteTypeText ]}];
+  [schema addObject:@{kMSETagColumnName : @[ kMSSQLiteTypeText ]}];
+  [schema addObject:@{kMSExpirationTimeColumnName : @[ kMSSQLiteTypeInteger ]}];
+  [schema addObject:@{kMSDownloadTimeColumnName : @[ kMSSQLiteTypeInteger ]}];
+  [schema addObject:@{kMSOperationTimeColumnName : @[ kMSSQLiteTypeInteger ]}];
+  [schema addObject:@{kMSPendingOperationColumnName : @[ kMSSQLiteTypeText ]}];
+  return schema;
```

Comment:
If you just want to insert items to the array and keep the immutability, you can do `NSArray *array = [mutableArray copy]` to convert it back to NSArray.

Reply:
https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1475/files#r272771158